translator/trace/protospan_to_jaegerthrift_test.go

```diff
@@ -0,0 +1,245 @@
+	}
+}
+
+func TestOCProtoToJaegerThrift(t *testing.T) {
+	const numOfFiles = 2
+	for i := 0; i < numOfFiles; i++ {
+		ocBatch := ocBatches[i]
+
+		gotJBatch, err := OCProtoToJaegerThrift(ocBatch)
+		if err != nil {
+			t.Errorf("Failed to translate OC batch to Jaeger Thrift: %v", err)
+			continue
+		}
+
+		wantSpanCount, gotSpanCount := len(ocBatch.Spans), len(gotJBatch.Spans)
+		if wantSpanCount != gotSpanCount {
+			t.Errorf("Different number of spans in the batches on pass #%d (want %d, got %d)", i, wantSpanCount, gotSpanCount)
+			continue
+		}
+
+		// Jaeger binary tags do not round trip from Jaeger -> OCProto -> Jaeger.
+		// For tests use data without binary tags.
+		thriftFile := fmt.Sprintf("./testdata/thrift_batch_no_binary_tags_%02d.json", i+1)
+		wantJBatch := &jaeger.Batch{}
+		if err := loadFromJSON(thriftFile, wantJBatch); err != nil {
+			t.Errorf("Failed load Jaeger Thrift from %q: %v", thriftFile, err)
+			continue
+		}
+
+		// Sort tags to help with comparison, not only for jaeger.Process but also
+		// on each span.
+		sort.Slice(gotJBatch.Process.Tags, func(i, j int) bool {
+			return gotJBatch.Process.Tags[i].Key < gotJBatch.Process.Tags[j].Key
+		})
+		sort.Slice(wantJBatch.Process.Tags, func(i, j int) bool {
+			return wantJBatch.Process.Tags[i].Key < wantJBatch.Process.Tags[j].Key
+		})
+		var jSpans []*jaeger.Span
+		jSpans = append(jSpans, gotJBatch.Spans...)
+		jSpans = append(jSpans, wantJBatch.Spans...)
+		for _, jSpan := range jSpans {
+			sort.Slice(jSpan.Tags, func(i, j int) bool {
+				return jSpan.Tags[i].Key < jSpan.Tags[j].Key
+			})
+		}
+
+		gjson, _ := json.Marshal(gotJBatch)
+		wjson, _ := json.Marshal(wantJBatch)
+		gjsonStr := testutils.GenerateNormalizedJSON(string(gjson))
+		wjsonStr := testutils.GenerateNormalizedJSON(string(wjson))
+		if gjsonStr != wjsonStr {
+			t.Errorf("OC Proto to Jaeger Thrift failed.\nGot:\n%s\nWant:\n%s\n", gjsonStr, wjsonStr)
+			t.Errorf("Saving formated jsons at ./testdata/ to help with further investigation")
+			testutils.SaveAsFormattedJSON(fmt.Sprintf("./testdata/failure_oc_to_jaeger_got_%02d.json", i), gotJBatch)
+			testutils.SaveAsFormattedJSON(fmt.Sprintf("./testdata/failure_oc_to_jaeger_want_%02d.json", i), wantJBatch)
+		}
+	}
+}
+
+// ocBatches has the OpenCensus proto batches used in the test. They are hard coded because
+// structs like tracepb.AttributeMap cannot be ready from JSON.
+var ocBatches = []*agenttracepb.ExportTraceServiceRequest{
+	{
+		Node: &commonpb.Node{
+			Identifier:  &commonpb.ProcessIdentifier{HostName: "api246-sjc1"},
+			LibraryInfo: &commonpb.LibraryInfo{ExporterVersion: "someVersion"},
+			ServiceInfo: &commonpb.ServiceInfo{Name: "api"},
+			Attributes: map[string]string{
+				"a.binary": "AQIDBAMCAQ==",
+				"a.bool":   "true",
+				"a.double": "1234.56789",
+				"a.long":   "123456789",
+				"ip":       "10.53.69.61",
+			},
+		},
+		Spans: []*tracepb.Span{
+			{
+				TraceId:      base64ToBytes("AAAAAAAAAABSlpqJVVcaPw=="),
```

Comment:
Why are we using base64? I think just plain bytes are easier to parse and easy to examine as a human and for example this is 24 bytes which becomes impossible to eyeball and catch conversion bugs. Please take a look at other tests for how we use plain `[]byte{0x01,....}`.